### PR TITLE
ath79:  add TP-Link WR842v3

### PR DESCRIPTION
**Pull request by SvenRoederer**
This adds support for the TP-Link WR842v3

* 16MB Flash; 64Mb RAM
* USB
* ath9k

Current status:
* WAN port working
* LAN ports (switch) ~~seem to not work yet~~ working
* WiFi ~~not tested~~ working (tested as Freifunk-node with AP and adhoc)
* USB ~~seems not to work, unsure if it's broken or a kernel-module is missing~~ working

### CODE REVIEW

File: target/linux/ath79/base-files/etc/board.d/01_leds [hunk 1]
```diff
@@ -154,7 +154,8 @@ tplink,tl-wr842n-v2)
 	ucidef_set_led_switch "lan4" "LAN4" "tp-link:green:lan4" "switch0" "0x02"
 	;;
 tplink,tl-wr841-v9|\
-tplink,tl-wr841-v11)
+tplink,tl-wr841-v11|\
+tplink,tl-wr841-v3)
```
Comment (adrianschmutzler) [inline target/linux/ath79/base-files/etc/board.d/01_leds]: 842n, not 841 here

File: target/linux/ath79/base-files/etc/board.d/02_network [hunk 2]
```diff
@@ -168,11 +168,17 @@ ath79_setup_interfaces()
...
 	ubnt,airrouter)
 		ucidef_set_interface_wan "eth0"
 		ucidef_add_switch "switch0" \
 			"0@eth1" "1:lan:4" "2:lan:3" "3:lan:2" "4:lan:1"
 		;;
+	tplink,tl-wr842n-v3dis)
+		ucidef_set_interface_wan "eth1"
+		ucidef_add_switch "switch0" \
+			"0@eth0" "1:lan:4" "2:lan:3" "3:lan:2" "4:lan:1"
+		;;
```
Comment (adrianschmutzler) [inline target/linux/ath79/base-files/etc/board.d/02_network]: This one for testing purposes? Or leftover?

**Comment by SvenRoederer** [comment]
fixed the LEDs and verified LAN-ports and WiFi

File: target/linux/ath79/dts/qca9533_tplink_tl-wr842n-v3.dts [hunk 3]
```diff
@@ -24,6 +24,7 @@
 			label = "tp-link:green:power";
 			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
 			default-state = "on";
+			linux,default-trigger = "heartbeat";
```
Comment (obsy) [inline target/linux/ath79/dts/qca9533_tplink_tl-wr842n-v3.dts]: This should be removed, also from 1043

**Comment by kofec** [comment]
Hi, thank for your work. I'm testing your image and seems to be fine till now.

File: target/linux/ath79/dts/qca9533_tplink_tl-wr842n-v3.dts (hunk 3 above)
Comment (SvenRoederer) [reply target/linux/ath79/dts/qca9533_tplink_tl-wr842n-v3.dts]: @obsy As using this feature should not be the default? Or is there any other reason? 
I personally like this feature.

**Comment by SvenRoederer** [comment]
@kofec Can you take a special look at the Wifi and QSS keys, if they work or not?

File: target/linux/ath79/dts/qca9533_tplink_tl-wr842n-v3.dts (hunk 3 above)
Comment (obsy) [reply target/linux/ath79/dts/qca9533_tplink_tl-wr842n-v3.dts]: Power led should be displayed on. If somebody want use heartbeat - it's easy to add by yourself in userland. We should not impose a led flashing. And openwrt does not use it by default, it tries to keep the leds working as in the original software.

**Comment by kofec** [comment]
It seems to be working.
When press "wifi" button - wifi was disable. Click again "wifi" button enabled/back wifi.
When press "reset/wps" - router was reboot .

**Comment by SvenRoederer** [comment]
@obsy removed the heartbeat
@kofec as mentioned, when using this board with our freifunk build-config, the buttons are working too. Can you give it a final test?
@dedeckeh as all functions are working, please remove the "WiP" tag and probably merge into master

**Comment by kofec** [comment]
Hi, All seems to be fine. Even USB:
```
[  306.400419] usb 1-1: new high-speed USB device number 2 using ehci-platform
[  306.636723] usb-storage 1-1:1.0: USB Mass Storage device detected
[  306.660507] scsi host0: usb-storage 1-1:1.0
[  307.702114] scsi 0:0:0:0: Direct-Access     USB      Flash Disk       8.07 PQ: 0 ANSI: 2
[  307.715599] sd 0:0:0:0: [sda] 3891200 512-byte logical blocks: (1.99 GB/1.86 GiB)
[  307.724614] sd 0:0:0:0: [sda] Write Protect is off
[  307.729609] sd 0:0:0:0: [sda] Mode Sense: 03 00 00 00
[  307.736112] sd 0:0:0:0: [sda] No Caching mode page found
[  307.741718] sd 0:0:0:0: [sda] Assuming drive cache: write through
[  308.016445]  sda: sda1
[  308.027580] sd 0:0:0:0: [sda] Attached SCSI removable disk
```

**Comment by chunkeey** [comment]
Merged. Thanks.

**Comment by adrianschmutzler** [comment]
Although this PR has been has been closed already, I would be interested in some additional data about the device(s) in case you still have access to it:

I'm currently working on a PR to store the label-mac-address in the DTS (https://github.com/openwrt/openwrt/pull/2159).

I would appreciate if you could provide information about your device:
Please run
`find /proc/device-tree/ -name "*mac-address*"`
on your device (just with the current firmware, no need to update anything; just has to be ath79, no ar71xx).
For each of the returned paths (if there are any), retrieve the mac-address with `cat`, e.g.
`cat /proc/device-tree/ahb/eth@19000000/mac-address`

Now check which of those returned MAC addresses matches the one on the label/sticker/case of your device and tell me the device-tree path leading to it. Note that you do not have to post your MAC addresses here, but just a statement about which match, e.g. like
```
/proc/device-tree/ahb/eth@1a000000/mac-address
/proc/device-tree/ahb/eth@1a000000/mtd-mac-address-increment
/proc/device-tree/ahb/eth@1a000000/mtd-mac-address
/proc/device-tree/ahb/eth@19000000/mac-address <== this one
/proc/device-tree/ahb/eth@19000000/mtd-mac-address
/proc/device-tree/ahb/apb/wmac@18100000/mac-address
/proc/device-tree/ahb/apb/wmac@18100000/mtd-mac-address
```

There may be one, two or no paths giving the correct address. It may also happen that the `find` command returns nothing. Thanks.

**Comment by adrianschmutzler** [comment]
Unfortunately, my description is slightly incorrect. The MAC addresses are stored binary, so you can't read with `cat`, but have to use the following code to read from the paths retrieved by `find`:
```
. /lib/functions/system.sh
get_mac_binary "$yourpath" 0
```
The include is obviously only required once. Sorry.

**Comment by SvenRoederer** [comment]
@adrianschmutzler 

> root@OpenWrt:/sys/firmware/devicetree/base/ahb# for file in eth@1*/*address; do ls -1 $file; . /lib/functions/system.sh; echo "  `get_mac_binary $file 0`"; done
eth@19000000/mac-address
  ec:08:6b:b8:21:13
eth@19000000/mtd-mac-address
  00:00:00:0a:00:01
eth@1a000000/mac-address
  ec:08:6b:b8:21:12
eth@1a000000/mtd-mac-address
  00:00:00:0a:00:01


matching address can be found in **eth@1a000000/mac-address**

**Comment by adrianschmutzler** [comment]
Thanks. So, `label-mac-device = &eth1;` is correct.

**Pull request closed without merging**